server/src/main/java/io/deephaven/server/console/completer/PythonAutoCompleteObserver.java

```diff
@@ -70,128 +70,87 @@ public void onNext(AutoCompleteRequest value) {
                 }
 
                 completer.callMethod("update_doc", document, uri, version);
-                break;
-            }
-            case GET_COMPLETION_ITEMS: {
-                GetCompletionItemsRequest request = value.getGetCompletionItems();
-                SessionState.ExportObject<ScriptSession> exportedConsole =
-                        session.getExport(request.getConsoleId(), "consoleId");
-                session.nonExport()
-                        .require(exportedConsole)
-                        .onError(responseObserver)
-                        .submit(() -> {
-                            getCompletionItems(request, exportedConsole, responseObserver);
-                        });
+                // TODO (https://github.com/deephaven/deephaven-core/issues/3614): Add publish diagnostics
+                // responseObserver.onNext(AutoCompleteResponse.newBuilder().setDiagnosticPublish());
                 break;
             }
             case CLOSE_DOCUMENT: {
-                CloseDocumentRequest request = value.getCloseDocument();
                 PyObject completer = (PyObject) scriptSession.get().getVariable("jedi_settings");
+                CloseDocumentRequest request = value.getCloseDocument();
                 completer.callMethod("close_doc", request.getTextDocument().getUri());
                 break;
             }
             case REQUEST_NOT_SET: {
                 throw Exceptions.statusRuntimeException(Code.INVALID_ARGUMENT, "Autocomplete command missing request");
             }
+            default: {
+                // Maintain compatibility with older clients
+                // The only autocomplete type supported before the consoleId was moved to the parent request was
+                // GetCompletionItems
+                final io.deephaven.proto.backplane.grpc.Ticket consoleId =
+                        value.hasConsoleId() ? value.getConsoleId() : value.getGetCompletionItems().getConsoleId();
```

Comment:
Should we have a safer version of this? If `console_id` is not set, ensure that request type is `get_completion_items`? (Could extract code and use on the java side too)